libs/remote_storage/src/s3_bucket.rs

```diff
@@ -201,12 +201,16 @@ impl S3Bucket {
     }
 
     fn relative_path_to_s3_object(&self, path: &RemotePath) -> String {
-        let mut full_path = self.prefix_in_bucket.clone().unwrap_or_default();
-        for segment in path.0.iter() {
-            full_path.push(REMOTE_STORAGE_PREFIX_SEPARATOR);
-            full_path.push_str(segment.to_str().unwrap_or_default());
+        assert_eq!(std::path::MAIN_SEPARATOR, REMOTE_STORAGE_PREFIX_SEPARATOR);
+        let path_string = path
+            .get_path()
+            .to_string_lossy()
+            .trim_end_matches(REMOTE_STORAGE_PREFIX_SEPARATOR)
```

Comment:
Why this line was necessary? I don't understand the logic of this change..

Reply:
https://neondb.slack.com/archives/C0277TKAJCA/p1690379585744019

Reply:
I don't remember. The unit tests (excpet for my new one testing this behavior) all pass without the `trim`.
On the other hand, this seems to more closely mirror the logic from before.